core/header.cpp

```diff
@@ -703,4 +709,131 @@ namespace MR
 
 
 
+
+
+
+
+  Header concatenate (const vector<Header>& headers, const size_t axis_to_concat, const bool permit_datatype_mismatch)
+  {
+    Exception e ("Unable to concatenate " + str(headers.size()) + " images along axis " + str(axis_to_concat) + ": ");
+
+    auto datatype_test = [&] (const bool condition)
+    {
+      if (condition && !permit_datatype_mismatch) {
+        e.push_back ("Mismatched data types");
+        throw e;
+      }
+      return condition;
+    };
+
+    auto concat_scheme = [] (Eigen::MatrixXd& existing, const Eigen::MatrixXd& extra)
+    {
+      if (!existing.rows())
+        return;
+      if (!extra.rows() || (extra.cols() != existing.cols())) {
+        existing.resize (0, 0);
+        return;
+      }
+      existing.conservativeResize (existing.rows() + extra.rows(), existing.cols());
```

Comment:
will likely need to cater for the possibility of different number of columns across the different inputs, and resize to the smaller of the two...

Reply:
Resizing only makes sense if it's guaranteed that a scheme with more columns will strictly be a superset of one with less columns. Not sure this would be the case; e.g. if storing the full *b*-tensor in `dw_scheme` I'd expect 6 values for the tensor then the *b*-value.